### ISSUE TITLE
feature: "folders" in bookmarks

### ISSUE THREAD

**Issue by corani** [lalitmee/browse.nvim#5]
Currently the bookmarks are a single flat list. By using nested lists it would be possible to support "folders". Something like:

```lua
bookmarks = {
    ["My first folder"] = {
        "https://some.nested/website",
    },
    "https://github.com/lalitmee/browse.nvim",
}
```

**Comment by lalitmee** [comment]
@corani, if you create folders in the `bookmarks` then how are you planning to show these in the `telescope` search?

**Comment by corani** [comment]
@lalitmee I'm not sure if Telescope supports some kind of drill-down, where after selecting the folder it shows a new dialog with the contents of that folder. If that is not the case, then perhaps the simplest thing to do is to treat it as a "filesystem", prefixing the links with their folder (something like `My first folder / https://some.nested/website`).

**Comment by lalitmee** [comment]
Implemented `aliases` or `tags` in ee0fc5d4ad66c35f0a02d917bc580de94262464e

**Comment by lalitmee** [comment]
For now, I have implemented `groups` support in `bookmarks` with 3336967066f57af56e6d2a8c07ac09acce351cd5. 

I don't think that I would like to have `folder` in my bookmarks. So if anyone is interested, please raise a PR.

I will keep this issue open.

**Comment by lalitmee** [comment]
@corani, I think the update which I have done yesterday to the `bookmarks` list, that solves your issue. Please try and let me know.

Then we can close this issue.
